ceph/ceph.py

```diff
@@ -153,22 +153,35 @@ def setup_ceph_firewall(self):
         Open required ports on nodes based on relevant ceph demons types
         """
         for node in self.get_nodes():
+            node.configure_firewall()
+
+            ports = None
             if node.role == "mon":
-                node.open_firewall_port(port="6789", protocol="tcp")
+                ports = ["6789"]
+
                 # for upgrades from 2.5 to 3.x, we convert mon to mgr
                 # so lets open ports from 6800 to 6820
-                node.open_firewall_port(port="6800-6820", protocol="tcp")
-            if node.role == "osd":
-                node.open_firewall_port(port="6800-7300", protocol="tcp")
-            if node.role == "mgr":
-                node.open_firewall_port(port="6800-6820", protocol="tcp")
-            if node.role == "mds":
-                node.open_firewall_port(port="6800", protocol="tcp")
-            if node.role == "iscsi-gw":
-                node.open_firewall_port(port="3260", protocol="tcp")
-                node.open_firewall_port(port="5000-5001", protocol="tcp")
-            if node.role == "grafana":
-                node.open_firewall_port(port="6800-6820", protocol="tcp")
+                ports.append("6800-6820")
+
+            elif node.role == "osd":
+                ports = "6800-7300"
+
+            elif node.role == "mgr":
+                ports = "6800-6820"
+
+            elif node.role == "mds":
+                ports = "6800"
+
+            elif node.role == "iscsi-gw":
+                ports = ["3260", "5000-5001"]
+
+            elif node.role == "grafana":
+                ports = "6800-6820"
+
+            else:
+                continue
+
+            node.open_firewall_port(port=ports, protocol="tcp")
```

Comment:
I believe this breaks the existing behavior. A node can have multiple roles i.e. Ceph services are co-located. In such a scenario `ports` holds only the last value of the satisfied conditional statement.

Reply:
@psathyan that was my first assumption but if you look at the current scenario, it iterating only once over the node list and exactly matching the `node.role` to word string.
If node has multiple roles, `node.list` should be a list or string with multiple words separated by comma. In this case if condition need to be changed as - 
```
if "mon" in node.role:
```

Please suggest.